src/SignalR/clients/ts/signalr/tests/HubConnection.test.ts

```diff
@@ -41,6 +41,97 @@ describe("HubConnection", () => {
             });
         });
 
+        it("can change url", async () => {
+            await VerifyLogger.run(async (logger) => {
+                const connection = new TestConnection();
+                const hubConnection = createHubConnection(connection, logger);
+                try {
+                    await hubConnection.start();
+                    expect(connection.sentData.length).toBe(1);
+                    expect(JSON.parse(connection.sentData[0])).toEqual({
+                        protocol: "json",
+                        version: 1,
+                    });
+                    const originalUrl = hubConnection.baseUrl;
+                    await hubConnection.stop();
+                    hubConnection.baseUrl = "http://newurl.com";
+                    expect(hubConnection.baseUrl).not.toBe(originalUrl);
+                    expect(hubConnection.baseUrl).toBe("http://newurl.com");
+                } finally {
+                    await hubConnection.stop();
+                }
+            });
+        });
+
+        it("can change url in onclose", async () => {
```

Comment:
Should we add a test to verify you cannot change the url in onreconnecting?

Reply:
I thought we did want to be able to change the while in the reconnecting event?

From the issue
> this should be doable in the Reconnecting event as well. @halter73 wants to try throwing InvalidOperationException if you try to set it while the connection is active (not reconnecting or disconnected).

Reply:
Then we should have a test for during reconnecting?

Reply:
Yeah my bad. We should add a test to verify you *can* change the url in onreconnecting.